mongo/options/clientencryptionoptions.go

```diff
@@ -29,6 +36,43 @@ func (c *ClientEncryptionOptions) SetKmsProviders(providers map[string]map[strin
 	return c
 }
 
+// SetTLSConfig applies custom TLS options to configure connections with each KMS provider.
+func (c *ClientEncryptionOptions) SetTLSConfig(tlsOptsMap map[string]map[string]interface{}) (*ClientEncryptionOptions, error) {
+	c.TLSConfig = make(map[string]tls.Config)
+
+	for provider, tlsOpts := range tlsOptsMap {
+		var cfg tls.Config
```

Comment:
The default `tls.Config` should enforce a minimum TLS version of v1.2.

E.g.
```go
cfg := &tls.Config{MinVersion: tls.VersionTLS12}
```

Reply:
Sounds good. Is there a way to require this in the structs for AutoEncryptionOptions and ClientEncryptionOptions or is this the only way to enforce min version 1.2?

Reply:
Good question. I think a similar change is required in `decryptKey` to use a default `tls.Config` with a minimum version if the map lookup does not have one. Left a separate comment with suggested change: https://github.com/mongodb/mongo-go-driver/pull/812/files#r753321847

Reply:
Got it, thanks for the suggested change